Improve queueStubsHelpers.js to avoid hacks and copy-pasted code

OleksandrChekhovskyi opened the pull request:
The existing implementation of apply() in `queueStubsHelpers.js` relied on hacking `_outstandingMethodBlocks`, but we can do it in a more clean way, with a special option to original apply. And no need to duplicate the logic of adding the method invoker to `_outstandingMethodBlocks` either.

Additionally, by accident, tests relied on the fact that the hack only touched `Meteor.connection` and not the actual connection it was invoked on. Added a check for `this._stream._neverQueued` in `apply()` like in other overridden functions here, to make it behave properly.

Also cleaning up some fibers leftovers, e.g. variable names and comments that no longer make any sense.

OleksandrChekhovskyi committed:
Do not check for applyAsyncPromise.stubPromise as it is always available

OleksandrChekhovskyi committed:
Clean up fibers-related leftovers in Connection._apply

OleksandrChekhovskyi committed:
Refactor queueStubsHelpers.js to avoid hacks and copy-pasted code

denihs commented:
The tests for this one are hanging. 

Could you please change the base branch to `release-3.0.4` and merge it into your branch? Then, the tests should be rerun.

OleksandrChekhovskyi commented:
> The tests for this one are hanging.
> 
> Could you please change the base branch to `release-3.0.4` and merge it into your branch? Then, the tests should be rerun.

Rebased on top of `release-3.0.4`, and switched the base branch.

OleksandrChekhovskyi commented:
Any comments, @nachocodoner, @denihs?

denihs reviewed (approved):
Nice! LGTM.

denihs commented:
I will squash and merge this one, as the Circle CI tests didn't run on your branch. If the tests fail, we can revert this merge.
